src/components/LogRow/AnsiiRow/index.tsx

```diff
@@ -11,9 +16,20 @@ const AnsiiRow = forwardRef<any, BaseRowProps>((rowProps, ref) => {
   // TODO: EVG-17535
   // This should be replaced with a collapsible component
   const lineIndex = Array.isArray(line) ? line[0] : line;
+  const lineContent = getLine(lineIndex);
+  const memoizedLogLine = useMemo(() => {
+    if (!lineContent) return null;
+    const render = linkifyHtml(ansiUp.ansi_to_html(lineContent), {
+      validate: {
+        url: (value: string) => /^(http)s?:\/\//.test(value),
+      },
+    });
+    return parse(render);
+  }, [lineContent]);
+  if (!lineContent) return null;
```

Comment:
I have two related questions:
1. Does `getLine` need to be typed as returning `string | undefined`? Because if not, I think we wouldn't need to check if `lineContent` is undefined
2. If it does need to be typed that way, do we need the line `if (!lineContent) return null;` twice?

Reply:
1. Yes because  in the chance we try to fetch a log line and we don't have its index we should return undefined. As a defensive tactic.
2. So `ansi_to_html` needs a value passed in so we need a guard to ensure we only pass in non null values. We also need a guard on line 29 to not return anything for a row if there isn't any content. That can probably change in the future. We can't move the guard up to after line 19 since that would break the rules of hooks. So it seemed the least hacky to just have 2 guards.